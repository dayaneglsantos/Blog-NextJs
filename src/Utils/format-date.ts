export const formatDate = (receivedDate: string) => {
  const date = new Date(receivedDate).toLocaleString('pt-BR', {
    timeStyle: 'short',
    dateStyle: 'short'
  })
  return date
}
