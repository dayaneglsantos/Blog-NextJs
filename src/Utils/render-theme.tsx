import { RenderResult, render } from '@testing-library/react'
import { ReactNode } from 'react'
import BlogThemeProvider from '../app/theme-provider'

export const renderTheme = (children: ReactNode): RenderResult => {
  return render(<BlogThemeProvider>{children}</BlogThemeProvider>)
}
