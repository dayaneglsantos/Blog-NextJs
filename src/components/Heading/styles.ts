'use client'
import styled from 'styled-components'
import { HeadingProps } from '.'
import theme from '../../Styles/theme'

export const Title = styled.h1<HeadingProps>`
  text-transform: ${(props) => (props.uppercase ? 'uppercase' : 'none')};
  text-align: center;
  color: ${({ theme }) => theme.colors.cyan};
  font-size: ${(props) =>
    props.size === 'small'
      ? theme.sizes.small
      : props.size === 'medium'
        ? theme.sizes.medium
        : props.size === 'big'
          ? theme.sizes.big
          : props.size === 'huge'
            ? theme.sizes.huge
            : ''};
`
