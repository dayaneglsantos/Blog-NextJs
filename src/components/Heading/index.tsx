import { Title } from './styles'

export type HeadingProps = {
  children: React.ReactNode
  dark?: boolean
  uppercase?: boolean
  as?: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6'
  size?: 'small' | 'medium' | 'big' | 'huge'
}

const Heading = ({
  children,
  dark = false,
  size = 'huge',
  as = 'h1',
  uppercase = false
}: HeadingProps) => {
  return (
    <Title as={as} size={size} dark={dark} uppercase={uppercase}>
      {children}
    </Title>
  )
}
export default Heading
