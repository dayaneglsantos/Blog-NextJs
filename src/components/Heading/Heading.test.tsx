import { screen } from '@testing-library/react'
import Heading from '.'
import { renderTheme } from '../../Utils/render-theme'

describe('<Heading />', () => {
  it('should render', () => {
    renderTheme(
      <Heading dark={false} as="h2" size="huge" uppercase={true}>
        Título
      </Heading>
    )
    const heading = screen.getByRole('heading', { name: 'Título' })
    expect(heading).toBeInTheDocument()
    expect(heading.tagName).toBe('H2')
  })
  it('should render wish default data', () => {
    renderTheme(<Heading>Título</Heading>)
    const heading = screen.getByRole('heading', { name: 'Título' })
    expect(heading).toBeInTheDocument()
    expect(heading.tagName).toBe('H1')
    expect(heading).toHaveStyleRule('font-size', '4.8rem')
  })
  it('should match snapshot', () => {
    const { container } = renderTheme(<Heading>Título</Heading>)
    expect(container).toMatchSnapshot()
  })
})
