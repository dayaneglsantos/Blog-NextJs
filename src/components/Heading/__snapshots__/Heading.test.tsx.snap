// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> should match snapshot 1`] = `
.c0 {
  text-transform: none;
  text-align: center;
  color: #008b8b;
  font-size: 4.8rem;
}

<div>
  <h1
    class="c0"
  >
    TÃ­tulo
  </h1>
</div>
`;
